server/routes/visitors.test.ts

```diff
@@ -465,6 +487,290 @@ describe('POST /visit/select-visitors', () => {
   })
 })
 
+describe('/visit/select-date-and-time', () => {
```

Comment:
Wrapped these in another level of `describe` and shuffled some of the test variables around a little to try and keep stuff closer to where it's needed - hopefully making file easier to eventually split up.